salt/modules/zypper.py

```diff
@@ -788,16 +789,23 @@ def mod_repo(repo, **kwargs):
 
     if kwargs.get('gpgautoimport') is True:
         global_cmd_opt.append('--gpg-auto-import-keys')
+        trigger_refresh = True
 
     if cmd_opt:
         cmd_opt = global_cmd_opt + ['mr'] + cmd_opt + [repo]
         __zypper__.refreshable.xml.call(*cmd_opt)
 
     # If repo nor added neither modified, error should be thrown
-    if not added and not cmd_opt:
+    if not added and not cmd_opt and not trigger_refresh:
```

Comment:
I would not forge such deranged shatter with unnecessary `trigger_refresh`, but would just:
```python
if added and kwargs.get('gpgautoimport'):
    refresh_opts = global_cmd_opt ...
else:
    raise CommandExecutionError( ....
```

That would look simple and clear.